01_marshmallow/2021_3/main.rb

```diff
@@ -0,0 +1,11 @@
+#!/bin/env ruby
+
+x = gets.chomp.split('').map(&:to_i)
+
+ans = x
+if x.include?(0)
+  idx = x.index(0)
+  ans = x[0..idx - 1] + x[idx..-1].map { |i| i = 1 }
```

Comment:
これがいいなら
```suggestion
  ans = x[0..idx - 1] + [1] * (x.size - idx - 1)
```
的な感じでもいける？(掛けてる個数が自信ない・・・)

Reply:
掛けてる個数は一つ足りないっぽいですかね 😌

Reply:
D言語だと `ans[idx..$] = 1` みたいな感じで書きたくなりますかねー。
Rubyだと範囲に対して1つの値をセットする代入文は無くて、下の形になるっぽかった。供養します。

```suggestion
  ans.fill(1, idx..)
```

Reply:
これがやりたかった...！
`fill`でいけるんですね。覚えておきます！

Reply:
`fill`便利すぎて感動
ループで回して変換するのと計算量は同じままなのでしょうか？

Reply:
> これがやりたかった...！

元の実装と違って fill でやると ans も x も値が変わる（同じ配列を参照してるので）ことには一応注意が必要です。
安全にやるなら `ans = x` を `ans = x.dup` にすると良いかな・・

> ループで回して変換するのと計算量は同じままなのでしょうか？

同じと思いますー。値を書き換える操作をする場合、その要素数分はどうしてもかかります。